src/assets/php8.4/php8.4.md

```diff
@@ -0,0 +1,226 @@
+---
+title: PHP 8.4 - Discover the Latest and Greatest
+authorName: Isha Baral
+authorAvatar: https://avatars.githubusercontent.com/u/40142570?s=400&u=58ce006ccc626577e503b75788b2e8634a9b7d9c&v=4
+authorLink: https://github.com/ishabaral/
+createdAt: Aug 22, 2024
+tags: php, php8.4, latest
+banner: https://blog.jankaritech.com/src/assets/php8.4/images/php8.4.png
+---
+
+Scheduled for release on November 21, 2024, PHP 8.4 packs some exciting new features and improvements. In this blog post, we'll explore the most interesting additions and changes:
+
+  1. New array helper functions
+  2. Property hooks
+  3. 'new' without parentheses
+  4. Implicitly nullable parameter declarations depreciated
+  5. New multibyte functions
+
+## 1. New array helper functions
+
+The following variants of array helper functions will be added in PHP 8.4:
+
+  * array_find()
+  * array_find_key()
+  * array_any()
+  * array_all()
+
+These functions will take an array and a callback function and return the following:
+
+
+| functions | Return value |
+| :---- | :---- |
+| array_find() | Returns the first element that meets the callback condition; `NULL` otherwise. |
+| array_find_key() | Returns the key of the first element that meets the callback condition; `NULL` otherwise. |
+| array_any() | Returns `true` if at least one element matches the callback condition; `false` otherwise. |
+| array_all() | Returns `true` if all elements match the callback condition; `false` otherwise. |
+
+ Note: `array_find()` retrieves only the first matching element. For multiple matches, consider using `array_filter()`.
+
+### Example
+
+Given an array with key-value pairs and a callback function:
+
+```
+$array = ['1'=> 'red', '2'=> 'purple', '3' => 'green']
+
+function hasLongName($value) {
+  return strlen($value) > 4;
+}
+```
+
+Here's how we can use the new functions:
+
+1. array_find:
+
+```
+// Find the first color with a name length greater than 4
+
+$result1 = array_find($array, 'hasLongName');
+
+var_dump($result1);  // string(5) "purple"
+```
+
+2. array_find_key:
+
+```
+// Find the key of the first color with a name length greater than 4
+
+$result2 = array_find_key($array, 'hasLongName');
+
+var_dump($result2);  // string(1) "2"
+```
+
+3. array_any:
+
+```
+// Check if any color name has a length greater than 4
+
+$result3 = array_any($array, 'hasLongName');
+
+var_dump($result3);  // bool(true)
+```
+
+4. array_all:
+
+```
+// Check if all color names have a length greater than 4
+
+$result4 = array_all($array, 'hasLongName');
+
+var_dump($result4);  // bool(false)
+```
+
+
```

Comment:
```suggestion
```
Adding a new line doesn't make any difference here while rendering. So, I think, we can delete this line.